components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1533,6 +1539,58 @@ public static String mapSignatureAlgorithm(String signatureAlgorithm) throws Ide
         return mapSignatureAlgorithmForJWSAlgorithm(signatureAlgorithm).getName();
     }
 
+    /**
+     * This method map encryption algorithm defined in identity.xml to nimbus
+     * encryption algorithm
+     *
+     * @param encryptionAlgorithm name of the encryption algorithm
+     * @return mapped JWEAlgorithm
+     * @throws IdentityOAuth2Exception
+     */
+    public static JWEAlgorithm mapEncryptionAlgorithmForJWEAlgorithm(String encryptionAlgorithm)
+            throws IdentityOAuth2Exception {
+
+        // TODO: 2/6/18 Refine
+        JWEAlgorithm jweAlgorithm = JWEAlgorithm.parse(encryptionAlgorithm);
+
+        if (jweAlgorithm.getRequirement() != null) {
+            return jweAlgorithm;
+        } else {
+            log.error("Unsupported Encryption Algorithm in identity.xml");
+            throw new IdentityOAuth2Exception("Unsupported Encryption Algorithm: " + encryptionAlgorithm + ", in identity.xml");
+        }
+
+//        if (NONE.equalsIgnoreCase(encryptionAlgorithm)) {
```

Comment:
Removed commented code

Reply:
I left it there to get comments in code review. I'll remove it after.